### ISSUE TITLE
storage.json structure

### ISSUE THREAD

**Issue by straighttalk86** [breakthestatic/firefly-teller-connector#1]
can you give an example storage.json structure, i figured out the teller_app_id part and the fireflyToken but couldn't quite figure out the structure of the accounts

**Comment by breakthestatic** [comment]
`storage.json` was intended to be populated by a web ui. I had created a basic settings page to add/update the app id, token, etc. I never pushed it up however, because I hadn't gotten around to implementing the ui for enabling/disabling individual accounts & handling unhealthy connections.

I just pushed up what I had sitting around locally, which provides the previously mentioned ui for the app id, token and importer url, although setting those up was easy enough as they are root level props. The actual account data is what comes from Teller, and is a lot more complex. To actually connect your accounts, you need to start the ui (e.g. `npm run connect`) and navigate to the running server, e.g. `http://localhost:8080`. From there, if you *dont* already have the teller app id populated (I did change the key name to better match the casing of the other keys), it should automatically redirect you to the settings page where you can enter that info. Once the required data is populated (either via the ui or manually editing the file), the home page has a "Connect" button, which starts the enrollment flow via Teller's SDK. Once you successfully complete an enrollment, it will save the required data structure to `storage.json` for you. By default, all newly added accounts are disabled though. To enable accounts (since I haven't created the ui yet), you will need to then manually edit `storage.json`, flipping the `enabled` flag for each account from `false` to `true`.

Hopefully that helps!  Also, sorry this isn't well-documented yet. I had originally scrambled to just get *something* up and running after finding out that Mint was shutting down and had meant to do a lot of clean-up before posting this publicly. However, I was switching computers at the time as well and figured just pushing my super rough code to GH would be easier than manually sending the project via usb stick, etc.

**Comment by straighttalk86** [comment]
Thanks for the very detailed response.  I just recently got into firefly setup and couldn't get plaid to work with my Chase account.  Teller is working, but it didn't have the auto import until I saw your project!  Even though I am a developer I don't find much time to code outside of work.  Thanks for the work so far.

**Comment by ohmnoms** [comment]
I appreciate your work on this. Also trying to set this up, running via Docker. Is the importer url for the Firefly Data Importer? I'm getting the Teller API to show up, with a successful connection to my bank, but I'm not seeing anything else in the storage.json besides the 3 initial config entries.

**Comment by breakthestatic** [comment]
> Is the importer url for the Firefly Data Importer? 

Yup, Importer URL is the URL used to access the Firefly Data Importer. It needs to include your importer secret in query params (it's been a while, not sure if that is part of the default URL or not). E.g. http://172.17.0.10/autoupload?secret=ABCDEFG

> I'm getting the Teller API to show up, with a successful connection to my bank, but I'm not seeing anything else in the storage.json besides the 3 initial config entries.

Once you complete the connection, the page should `POST` to `/api/enrollments`. Do you see that call in dev tools after saving the connection in the Teller.io flow?

**Comment by ohmnoms** [comment]
> Do you see that call in dev tools after saving the connection in the Teller.io flow?

No, but that helps. I think it has something to do with the importer connection and the way it's set up. There's some documentation that suggests confidential secrets won't [work with the data importer](https://docs.firefly-iii.org/how-to/firefly-iii/features/api/), is the secret necessary? I'm not seeing the post with or without it, but also can't authorize with the secret from the data importer tool

**Comment by breakthestatic** [comment]
The teller connector script doesn't directly use the API, it uses the HTTP POST method for importing data (see https://docs.firefly-iii.org/how-to/data-importer/advanced/post/ for info). The secret there is different from the Firefly III core token. It's set directly in the importer config. 

The whole process is a bit convoluted, as I ported my existing manual workflow into this script and didn't really think things through from the ground up (e.g. by using the actual API directly)

Hope that helps clarify the import process a bit at least.
